### PR TITLE
resource alicloud_slb_listener support acl.

### PR DESCRIPTION
**Pull request by wzb56**
TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudSlbListener -timeout=300m
=== RUN   TestAccAlicloudSlbListenersDataSource_http
--- PASS: TestAccAlicloudSlbListenersDataSource_http (37.85s)
=== RUN   TestAccAlicloudSlbListenersDataSource_tcp
--- PASS: TestAccAlicloudSlbListenersDataSource_tcp (32.91s)
=== RUN   TestAccAlicloudSlbListenersDataSource_udp
--- PASS: TestAccAlicloudSlbListenersDataSource_udp (36.42s)
=== RUN   TestAccAlicloudSlbListener_importHttp
--- PASS: TestAccAlicloudSlbListener_importHttp (14.38s)
=== RUN   TestAccAlicloudSlbListener_importTcp
--- PASS: TestAccAlicloudSlbListener_importTcp (13.94s)
=== RUN   TestAccAlicloudSlbListener_importUdp
--- PASS: TestAccAlicloudSlbListener_importUdp (14.85s)
=== RUN   TestAccAlicloudSlbListener_http
--- PASS: TestAccAlicloudSlbListener_http (15.34s)
=== RUN   TestAccAlicloudSlbListener_tcp
--- PASS: TestAccAlicloudSlbListener_tcp (11.98s)
=== RUN   TestAccAlicloudSlbListener_udp
--- PASS: TestAccAlicloudSlbListener_udp (13.57s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	191.324s

### CODE REVIEW

File: alicloud/resource_alicloud_slb_listener.go [hunk 1]
```diff
@@ -82,6 +82,21 @@ func resourceAliyunSlbListener() *schema.Resource {
...
+			},
+			"acl_type": &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validateAllowedStringValue([]string{string(AclTypeBlack), string(AclTypeWhite)}),
+				Optional:     true,
+			},
+			"acl_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_slb_listener.go]: I think `acl_type` and `acl_status` are valid when `acl_id` is set. So,  `acl_type` and `acl_status` need a DiffSuppressFunc

File: alicloud/resource_alicloud_slb_listener_test.go [hunk 2]
```diff
@@ -29,6 +29,10 @@ func TestAccAlicloudSlbListener_http(t *testing.T) {
 						"alicloud_slb_listener.http", "protocol", "http"),
 					resource.TestCheckResourceAttr(
 						"alicloud_slb_listener.http", "backend_port", "80"),
+					resource.TestCheckResourceAttr(
+						"alicloud_slb_listener.http", "acl_status", "on"),
+					resource.TestCheckResourceAttr(
+						"alicloud_slb_listener.http", "acl_type", string(AclTypeWhite)),
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_slb_listener_test.go]: Please check aclI_id using `TestCheckResourceAttrSet`

File: alicloud/resource_alicloud_slb_listener.go [hunk 3]
```diff
@@ -82,6 +82,21 @@ func resourceAliyunSlbListener() *schema.Resource {
...
+			"acl_status": &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validateAllowedStringValue([]string{string(OnFlag), string(OffFlag)}),
+				Optional:     true,
+				Default:      OffFlag,
+			},
+			"acl_type": &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validateAllowedStringValue([]string{string(AclTypeBlack), string(AclTypeWhite)}),
+				Optional:     true,
```
[the remark this answers is not in the document]
Comment (wzb56) [reply alicloud/resource_alicloud_slb_listener.go]: when acl_status='on' then acl_type/acl_id is required.

**Comment by wzb56** [comment]
zhongbo.wzb@scala:~/go/src/github.com/terraform-providers/terraform-provider-alicloud% TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudSlbListener -timeout=300m -count=1
=== RUN   TestAccAlicloudSlbListenersDataSource_http
--- PASS: TestAccAlicloudSlbListenersDataSource_http (40.49s)
=== RUN   TestAccAlicloudSlbListenersDataSource_tcp
--- PASS: TestAccAlicloudSlbListenersDataSource_tcp (36.53s)
=== RUN   TestAccAlicloudSlbListenersDataSource_udp
--- PASS: TestAccAlicloudSlbListenersDataSource_udp (32.98s)
=== RUN   TestAccAlicloudSlbListener_importHttp
--- PASS: TestAccAlicloudSlbListener_importHttp (13.40s)
=== RUN   TestAccAlicloudSlbListener_importTcp
--- PASS: TestAccAlicloudSlbListener_importTcp (12.75s)
=== RUN   TestAccAlicloudSlbListener_importUdp
--- PASS: TestAccAlicloudSlbListener_importUdp (12.14s)
=== RUN   TestAccAlicloudSlbListener_http
--- PASS: TestAccAlicloudSlbListener_http (12.05s)
=== RUN   TestAccAlicloudSlbListener_tcp
--- PASS: TestAccAlicloudSlbListener_tcp (11.86s)
=== RUN   TestAccAlicloudSlbListener_udp
--- PASS: TestAccAlicloudSlbListener_udp (13.12s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	185.442s

**Comment by wzb56** [comment]
zhongbo.wzb@scala:~/go/src/github.com/terraform-providers/terraform-provider-alicloud% TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudSlbListener -timeout=300m

^@=== RUN   TestAccAlicloudSlbListenersDataSource_http
^@^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_http (40.70s)
=== RUN   TestAccAlicloudSlbListenersDataSource_tcp
^@^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_tcp (32.48s)
=== RUN   TestAccAlicloudSlbListenersDataSource_udp
^@^@^@--- PASS: TestAccAlicloudSlbListenersDataSource_udp (33.73s)
=== RUN   TestAccAlicloudSlbListener_importHttp
^@--- PASS: TestAccAlicloudSlbListener_importHttp (12.55s)
=== RUN   TestAccAlicloudSlbListener_importTcp
^@^@--- PASS: TestAccAlicloudSlbListener_importTcp (21.57s)
=== RUN   TestAccAlicloudSlbListener_importUdp
^@^@--- PASS: TestAccAlicloudSlbListener_importUdp (12.18s)
=== RUN   TestAccAlicloudSlbListener_http
^@--- PASS: TestAccAlicloudSlbListener_http (15.14s)
=== RUN   TestAccAlicloudSlbListener_tcp
^@^@--- PASS: TestAccAlicloudSlbListener_tcp (16.09s)
=== RUN   TestAccAlicloudSlbListener_udp
^@--- PASS: TestAccAlicloudSlbListener_udp (13.85s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	198.357s

**Pull request merged**